x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -369,6 +395,48 @@ export const reindexServiceFactory = (
   // ------ The service itself
 
   return {
+    async hasRequiredPrivileges(indexName: string) {
+      // If security is disabled, return true.
+      const xpackInfo = await callCluster('transport.request', {
+        path: '/_xpack',
+        method: 'GET',
+      });
+      if (!get(xpackInfo, 'features.security.enabled')) {
```

Comment:
We need to make sure it's both enabled and available. An example of this is if the license is expired.